java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java

```diff
@@ -291,6 +275,88 @@ public Optional<BulkDataRules> getApplicableRules(RuleSet rules, String sourceOb
         return Optional.ofNullable(applicableRules);
     }
 
+
+    /**
+     * validate that the input stream can be processed per request/transform
+     *
+     * use-case: avoid opening output writer if file isn't valid
+     *
+     * @param request
+     * @param transform
+     * @param inputStreamSupplier
+     *
+     * @return
+     */
+    @SneakyThrows
+    boolean validate(StorageEventRequest request,
+                     StorageHandler.ObjectTransform transform,
+                     Supplier<InputStream> inputStreamSupplier) {
+        int bufferSize = getBufferSize();
+        try (
+            InputStream decompressedStream = request.getDecompressInput() ? new GZIPInputStream(inputStreamSupplier.get(), bufferSize) : inputStreamSupplier.get();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(decompressedStream, StandardCharsets.UTF_8), bufferSize);
+            OutputStream out = new ByteArrayOutputStream()
+        ) {
+
+            StringBuffer firstLines = new StringBuffer();
+            while (firstLines.length() < 5) {
```

Comment:
This is mostly for checking if the buffer has less than 5 characters length; maybe I'd change this to `firstLine` and just do the break after reading it or if the idea is to read first 5 lines, just count the lines read instead of `length()`

Reply:
copilot fail